crypto/signature.go

```diff
@@ -305,7 +297,7 @@ func createVerifyResult(
 			signatureError = newSignatureNoVerifier()
 		case signature.SignatureError != nil:
 			signatureError = newSignatureFailed(signature.SignatureError)
-		case signature.CorrespondingSig == nil || !isHashAllowed(signature.CorrespondingSig.Hash):
+		case signature.CorrespondingSig == nil:
 			signatureError = newSignatureInsecure()
```

Comment:
I think this entire case and the `newSignatureInsecure()` error can be removed entirely

Reply:
The context check afterward could panic with a null pointer exceptions if we are not checking for nil. However, we could move it to signature the context check logic to make it clearer?

Reply:
Yeah, I'd do that :+1:

Reply:
Thanks for the input, fixed with:
https://github.com/ProtonMail/gopenpgp/pull/303/commits/274f52599ea3a90f8b3d2742d266cf48f780250c